gen/string.go

```diff
@@ -25,8 +25,19 @@ import (
 	"strings"
 	"unicode"
 	"unicode/utf8"
+
+	"go.uber.org/thriftrw/compile"
 )
 
+// ItemSpec is any Thrift compile items with a name and annotations.
+type ItemSpec interface {
```

Comment:
discussed offline:
 - renamed to namedEntity, private interface to strings.go
 - TypeSpec doesn't require ThriftAnnotations as not all TypeSpec are offering annotations.

Composition wins!